command/connect/expose/expose.go

```diff
@@ -0,0 +1,238 @@
+package expose
+
+import (
+	"flag"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/consul/agent"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/command/flags"
+	"github.com/hashicorp/consul/command/intention/create"
+	"github.com/hashicorp/consul/command/intention/finder"
+	"github.com/mitchellh/cli"
+)
+
+func New(ui cli.Ui) *cmd {
+	c := &cmd{UI: ui}
+	c.init()
+	return c
+}
+
+type cmd struct {
+	UI    cli.Ui
+	flags *flag.FlagSet
+	http  *flags.HTTPFlags
+	help  string
+
+	// flags
+	ingressGateway string
+	service        string
+	portRaw        string
+	port           int
+	protocol       string
+}
+
+func (c *cmd) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flags.StringVar(&c.ingressGateway, "ingress-gateway", "",
+		"(Required) The name of the ingress gateway service to use. A namespace "+
+			"can optionally be specified as a prefix via the 'namespace/service' format.")
+
+	c.flags.StringVar(&c.service, "service", "",
+		"(Required) The name of destination service to expose. A namespace "+
+			"can optionally be specified as a prefix via the 'namespace/service' format.")
+
+	c.flags.StringVar(&c.portRaw, "port", "",
```

Comment:
Is there a reason we cannot use `c.flags.IntVar` here instead, so that we can avoid the parsing of port later on?